File: fix_ui.js
const fs = require('fs');
const glob = require('glob'); // Note: we can just use native fs since it's a small codebase
const path = require('path');

const srcDir = path.join(__dirname, 'frontend', 'src');

function fixFlex() {
    const files = fs.readdirSync(srcDir).filter(f => f.endsWith('.jsx'));
    let changedFiles = 0;
    for (const f of files) {
        const fp = path.join(srcDir, f);
        let content = fs.readFileSync(fp, 'utf-8');
        let original = content;

        // Find standard flex space-between rows and add flexWrap wrap if not present
        // Looking for variations of display: "flex", justifyContent: "space-between"
        const rg = /display\s*:\s*["']flex["'][^}]*justifyContent\s*:\s*["']space-between["']/g;
        let match;
        const toReplace = [];
        while ((match = rg.exec(content)) !== null) {
            const block = match[0];
            if (!block.includes('flexWrap')) {
                // Add flexWrap and gap (if gap is missing)
                let newBlock = block + ', flexWrap: "wrap"';
                if (!block.includes('gap')) {
                    newBlock += ', gap: 10';
                }
                toReplace.push({ old: block, new: newBlock });
            }
        }

        // Deduplicate
        const uniqueReplacements = [...new Set(toReplace.map(o => JSON.stringify(o)))].map(s => JSON.parse(s));
        
        for (const repl of uniqueReplacements) {
            content = content.split(repl.old).join(repl.new);
        }

        if (content !== original) {
            fs.writeFileSync(fp, content, 'utf-8');
            changedFiles++;
            console.log(`Updated flex containers in ${f}`);
        }
    }
    console.log(`Fixed flex in ${changedFiles} files`);
}

function fixGridAtom() {
    const fp = path.join(srcDir, 'gap.atoms.jsx');
    let content = fs.readFileSync(fp, 'utf-8');
    
    // Add media query
    if (!content.includes('.gap-grid-responsive')) {
        content = content.replace(
            /export const GLOBAL_CSS = `/,
            "export const GLOBAL_CSS = `\n@media (max-width: 768px) {\n  .gap-grid-responsive { grid-template-columns: 1fr !important; }\n}\n"
        );
    }

    // Change Grid component
    const oldGridRegex = /<div style={{\s*display:\s*"grid",\s*gridTemplateColumns:[^}]+,\s*gap,\s*}}>\{children\}<\/div>/;
    const newGridStr = `<div className="gap-grid-responsive" style={{
    display: "grid",
    gridTemplateColumns: cols ? \`repeat(\${cols}, minmax(0, 1fr))\` : \`repeat(auto-fill, minmax(\${minCol}, 1fr))\`,
    gap,
  }}>{children}</div>`;

    let replaced = false;
    content = content.replace(/export const Grid = \(\{([^}]+)\}\) => \(([\s\S]*?)\);/, (match, p1, p2) => {
        replaced = true;
        return `export const Grid = ({${p1}}) => (\n  ${newGridStr}\n);`;
    });

    if (replaced || content.includes('.gap-grid-responsive')) {
        fs.writeFileSync(fp, content, 'utf-8');
        console.log(`Updated gap.atoms.jsx Grid to use reliable media queries.`);
    }
}

fixGridAtom();
fixFlex();

File: fix_ui.py
import os
import re

src_dir = os.path.join(os.path.dirname(__file__), 'frontend', 'src')

def fix_flex():
    changed_files = 0
    for f in os.listdir(src_dir):
        if f.endswith('.jsx'):
            fp = os.path.join(src_dir, f)
            with open(fp, 'r', encoding='utf-8') as file:
                content = file.read()
            original = content
            
            # Find display: "flex", justifyContent: "space-between" and replace
            rg = re.compile(r'display\s*:\s*["\']flex["\'][^}]+?justifyContent\s*:\s*["\']space-between["\']')
            matches = rg.findall(content)
            
            for block in set(matches):
                if 'flexWrap' not in block:
                    new_block = block + ', flexWrap: "wrap"'
                    if 'gap' not in block:
                        new_block += ', gap: 10'
                    content = content.replace(block, new_block)
            
            if content != original:
                with open(fp, 'w', encoding='utf-8') as file:
                    file.write(content)
                changed_files += 1
                print(f"Updated flex containers in {f}")
    print(f"Fixed flex in {changed_files} files")

def fix_grid_atom():
    fp = os.path.join(src_dir, 'gap.atoms.jsx')
    with open(fp, 'r', encoding='utf-8') as file:
        content = file.read()
        
    original = content
    
    if '.gap-grid-responsive' not in content:
        content = content.replace('export const GLOBAL_CSS = `', "export const GLOBAL_CSS = `\n@media (max-width: 768px) {\n  .gap-grid-responsive { grid-template-columns: 1fr !important; }\n}\n")
        
    new_grid = """<div className="gap-grid-responsive" style={{
    display: "grid",
    gridTemplateColumns: cols ? `repeat(${cols}, minmax(0, 1fr))` : `repeat(auto-fill, minmax(${minCol}, 1fr))`,
    gap,
  }}>{children}</div>"""
  
    content = re.sub(r'export const Grid = \(\{([^}]+)\}\) => \(([\s\S]*?)\);', 
                     lambda m: f"export const Grid = ({{{m.group(1)}}}) => (\n  {new_grid}\n);", 
                     content)
                     
    if content != original:
        with open(fp, 'w', encoding='utf-8') as file:
            file.write(content)
        print("Updated gap.atoms.jsx Grid to use reliable media queries.")

fix_grid_atom()
fix_flex()
